sklearn/discriminant_analysis.py

```diff
@@ -120,11 +120,11 @@ def _class_cov(X, y, priors=None, shrinkage=None):
         Class covariance matrix.
     """
     classes = np.unique(y)
-    covs = []
-    for group in classes:
+    cov = np.zeros(shape=(X.shape[1], X.shape[1]))
+    for idx, group in enumerate(classes):
         Xg = X[y == group, :]
-        covs.append(np.atleast_2d(_cov(Xg, shrinkage)))
-    return np.average(covs, axis=0, weights=priors)
+        cov += priors[idx] * np.atleast_2d(_cov(Xg, shrinkage))
+    return cov
```

Comment:
In your version, you don't return the weighted average as before, but the weighted sum.
You should divide `cov` by the sum of  `priors`, or by `n_classes` if the priors are `None`.

Reply:
Ok you answered it already, my mistake